### PR TITLE
Install Script: Add comments for nearly every line of code

### PR DESCRIPTION
**Pull request by jacobsalmela**
**By submitting this pull request, I confirm the following (please check boxes, eg [X]) _Failure to fill the template will close your PR_:**

***Please submit all pull requests against the `development` branch. Failure to do so will delay or deny your request***

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] I have checked that [another pull request](https://github.com/pi-hole/pi-hole/pulls) for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:**

5
---
This pull adds comments to nearly every line in the installer.  As with the new debug script, the intent of this pull is to:
- help the developer team determine where code improvements can be made
- promote a culture of documentation within the code via comments
- make Pi-hole more appealing to contributors because they will be able to better understand why the code is the way that it is
- make Pi-hole more appealing to beginners of open source who want to contribute or are just learning about projects (such as the Raspberry Pi)
- promote better `bash` coding practices such as functions that do one thing, defensive programming, exit codes, etc.

_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

### CODE REVIEW

File: automated install/basic-install.sh [hunk 1]
```diff
@@ -97,57 +125,80 @@ show_ascii_berry() {
 
 # Compatibility
 distro_check() {
+# If apt-get is isntalled, then we know it's part of the Debian family
```
Comment (dxm765) [inline automated install/basic-install.sh]: Spelling error, installed.

File: automated install/basic-install.sh [hunk 2]
```diff
@@ -464,39 +645,50 @@ It is also possible to use a DHCP reservation, but if you are going to do that,
   fi
 }
 
+# dhcpcd is very annoying,
 setDHCPCD() {
-  # Append these lines to dhcpcd.conf to enable a static IP
+  # but we cab append these lines to dhcpcd.conf to enable a static IP
```
Comment (dxm765) [inline automated install/basic-install.sh]: Spelling error, can.

File: automated install/basic-install.sh (hunk 1 above)
Comment (jacobsalmela) [inline automated install/basic-install.sh]: Fixed, thanks.

File: automated install/basic-install.sh (hunk 2 above)
Comment (jacobsalmela) [inline automated install/basic-install.sh]: Fixed, thanks.

File: automated install/basic-install.sh [hunk 3]
```diff
@@ -97,57 +125,80 @@ show_ascii_berry() {
...
   if ${PKG_MANAGER} install --dry-run php > /dev/null 2>&1; then
     phpVer="php"
+  # If not,
   else
+    # fall back on the php5 packages
     phpVer="php5"
   fi
-  # #########################################
+
+  # Since our instal script is so large, we need several other programs to successfuly get a machine provisioned
```
Comment (dxm765) [inline automated install/basic-install.sh]: Spelling mistake , install script.

File: automated install/basic-install.sh [hunk 4]
```diff
@@ -162,113 +213,176 @@ elif command -v rpm &> /dev/null; then
...
 }
 
+# A function for checking if a folder is a git repository
 is_repo() {
-  # Use git to check if directory is currently under VCS, return the value 128
-  # if directory is not a repo. Return 1 if directory does not exist.
+  # Use a named, local variable instead of the vague $1, which is the first arguement passed to this function
+  # These local variables should always be lowercase
   local directory="${1}"
+  # A local variabled for the current directory
```
Comment (dxm765) [inline automated install/basic-install.sh]: Possible grammar mistake, variable.

File: automated install/basic-install.sh [hunk 5]
```diff
@@ -632,117 +870,157 @@ setDNS() {
...
         sed -i 's/^log-queries/#log-queries/' ${dnsmasq_pihole_01_location}
+    # Otherwise,
     else
-        #Enable Logging
+        # enable it by uncommenting the directive in the DNS config file
         sed -i 's/^#log-queries/log-queries/' ${dnsmasq_pihole_01_location}
     fi
 }
 
+# Clean an exiting installation to prepare for upgrade/reinstall
```
Comment (dxm765) [inline automated install/basic-install.sh]: Spelling error, clean existing.

File: automated install/basic-install.sh (hunk 3 above)
Comment (jacobsalmela) [inline automated install/basic-install.sh]: Fixed.

File: automated install/basic-install.sh (hunk 4 above)
Comment (jacobsalmela) [inline automated install/basic-install.sh]: Fixed.

File: automated install/basic-install.sh (hunk 5 above)
Comment (jacobsalmela) [inline automated install/basic-install.sh]: Fixed.

**Comment by dschaper** [comment]
@jacobsalmela Just need to pull in the changes from the Development branch and this should be good to merge.

File: automated install/basic-install.sh [hunk 6]
```diff
@@ -895,178 +1217,230 @@ install_dependent_packages() {
   # amount of download traffic.
   # NOTE: We may be able to use this installArray in the future to create a list of package that were
   # installed by us, and remove only the installed packages, and not the entire list.
-  if command -v debconf-apt-progress &> /dev/null; then
+  if command -v debconf-apt-progress &> /dev/null;
```
Comment (dschaper) [inline automated install/basic-install.sh]: missing a `then`

**Comment by pralor** [comment]
This pull request has been mentioned on **Pi-hole Userspace**. There might be relevant details there:

https://discourse.pi-hole.net/t/pi-hole-v3-2-introduces-long-term-statistics-an-audit-log-colours-and-more/5772/1

**Pull request merged**